# IS362 - Project Two

Choose any three of the “wide” datasets identified in the Week 5 Discussion items. For each of the three chosen datasets:
• Create a .CSV file (or optionally, a MySQL database!) that includes all of the information
included in the dataset. You’re encouraged to use a “wide” structure similar to how the
information appears in the discussion item, so that you can practice tidying and
transformations as described below.
• Read the information from your .CSV file into Python, and use pandas as needed to tidy
and transform your data. [Most of your grade will be based on this step!]
• Perform the analysis requested in the discussion item.
• Your code should be in an Jupyter Notebook, posted to your GitHub repository, and
should include narrative descriptions of your data cleanup work, analysis, and
conclusions. 

The three data sets which I have selected for this project are the California Cities data set, the NYC Cause of Death data set, and the Student data set. I have added all three data sets to the repository.

Let's begin with the imports we'll need to begin cleaning and analyzing this data:

In [1]:
import pandas as pd
import numpy as np

# California Cities Data Analysis

Let's start with the California Cities data set. This data set contains information on a variety of cities in the state of California which includes their name, coordinates, elevation, population, total area, and the percentages of land area vs water area. 

In [2]:
calidata = pd.read_csv("C:/Users/Jessica/Desktop/John's School/IS362/IS362_Project2/california_cities_data.csv")
calidata

Unnamed: 0             city       latd       longd  elevation_m  \
0             0         Adelanto  34.576111 -117.432778        875.0   
1             1      AgouraHills  34.153333 -118.761667        281.0   
2             2          Alameda  37.756111 -122.274444          NaN   
3             3           Albany  37.886944 -122.297778          NaN   
4             4         Alhambra  34.081944 -118.135000        150.0   
5             5       AlisoViejo  33.575000 -117.725556        127.0   
6             6          Alturas  41.487222 -120.542500       1332.0   
7             7       AmadorCity  38.419444 -120.824167        280.0   
8             8   AmericanCanyon  38.168056 -122.252500         14.0   
9             9          Anaheim  33.836111 -117.889722         48.0   
10           10         Anderson  40.452222 -122.296667        132.0   
11           11       AngelsCamp  38.068333 -120.539722        420.0   
12           12          Antioch  38.005000 -121.805833         13.0   
13           13      AppleValley  34.516667 -117.216667        898.0   
14           14          Arcadia  34.132778 -118.036389        147.0   
15           15           Arcata  40.866389 -124.082778          NaN   
16           16     ArroyoGrande  35.120833 -120.586667         36.0   
17           17          Artesia  33.867222 -118.080556         16.0   
18           18            Arvin  35.209167 -118.828333        137.0   
19           19       Atascadero  35.484167 -120.672500        268.0   
20           20         Atherton  37.458611 -122.200000         18.0   
21           21          Atwater  37.347778 -120.609167         46.0   
22           22           Auburn  38.898611 -121.074444        374.0   
23           23           Avalon  33.340833 -118.327778          9.0   
24           24           Avenal  36.004167 -120.128889        246.0   
25           25            Azusa  34.130556 -117.906944        186.0   
26           26      Bakersfield  35.366667 -119.016667          NaN   
27           27      BaldwinPark  34.082778 -117.971667        114.0   
28           28          Banning  33.931667 -116.897500        716.0   
29           29          Barstow  34.900000 -117.016667        664.0   
..          ...              ...        ...         ...          ...   
452         452            Vista  33.193611 -117.241111         99.0   
453         453           Walnut  34.033333 -117.866667        171.0   
454         454      WalnutCreek  37.906389 -122.065000          NaN   
455         455            Wasco  35.594167 -119.340833        100.0   
456         456        Waterford  37.645000 -120.767500         52.0   
457         457      Watsonville  36.920000 -121.763611          9.0   
458         458             Weed  41.424167 -122.384444       1044.0   
459         459       WestCovina  34.056667 -117.918611        110.0   
460         460    WestHollywood  34.087778 -118.372222         86.0   
461         461  WestlakeVillage  34.141944 -118.819444        268.0   
462         462      Westminster  33.751389 -117.993889         12.0   
463         463      Westmorland  33.037222 -115.621389          NaN   
464         464   WestSacramento  38.580556 -121.530278          6.0   
465         465        Wheatland  39.010000 -121.423056         28.0   
466         466         Whittier  33.965556 -118.024444        112.0   
467         467         Wildomar  33.603611 -117.272778        387.0   
468         468         Williams  39.154722 -122.149444         25.0   
469         469          Willits  39.409722 -123.355556        424.0   
470         470          Willows  39.524444 -122.193611         42.0   
471         471          Windsor  38.546111 -122.805278         36.0   
472         472          Winters  38.525000 -121.970833         40.0   
473         473         Woodlake  36.416389 -119.099444        134.0   
474         474         Woodland  38.678611 -121.773333         21.0   
475         475         Woodside  37.420833 -

While we can see that the data in this data set is not necessarily poorly organized, it does have a lot of null values and zero values, so any analysis we want to do on this data will require that we address that. We also see that the first column has no name or explanation for the data it contains. They appear to be unique identifiers for the cities, so let's start by giving that column a descriptive name.

In [3]:
calidata.columns.values[0] = 'city_id'
calidata.head()

city_id         city       latd       longd  elevation_m  elevation_ft  \
0        0     Adelanto  34.576111 -117.432778        875.0        2871.0   
1        1  AgouraHills  34.153333 -118.761667        281.0         922.0   
2        2      Alameda  37.756111 -122.274444          NaN          33.0   
3        3       Albany  37.886944 -122.297778          NaN          43.0   
4        4     Alhambra  34.081944 -118.135000        150.0         492.0   

   population_total  area_total_sq_mi  area_land_sq_mi  area_water_sq_mi  \
0             31765            56.027           56.009             0.018   
1             20330             7.822            7.793             0.029   
2             75467            22.960           10.611            12.349   
3             18969             5.465            1.788             3.677   
4             83089             7.632            7.631             0.001   

   area_total_km2  area_land_km2  area_water_km2  area_water_percent  
0         145.107        145.062           0.046                0.03  
1          20.260         20.184           0.076                0.37  
2          59.465         27.482          31.983               53.79  
3          14.155          4.632           9.524               67.28  
4          19.766         19.763           0.003                0.01

Let's ask a few questions about the data:
• What city has the highest and lowest elevation? 
• What city has the highest and lowest population?
• What city has the greatest total area?

We'll start by looking at the cities with the highest and lowest elevation. We'll do this by sorting the data according to the elevation column. Although the data set provides the elevation in both meters and feet, we're going to use feet because that column has fewer null values. 

Here are the five cities with the highest elevation:

In [4]:
calidata[["city", "elevation_ft"]].sort_values(by=["elevation_ft"], ascending=False).head()

city  elevation_ft
246    MammothLakes        7880.0
39      BigBearLake        6752.0
416  SouthLakeTahoe        6237.0
436         Truckee        5817.0
242        Loyalton        4951.0

Here are the five cities with the lowest elevation: 

In [5]:
calidata[["city", "elevation_ft"]].sort_values(by=["elevation_ft"], ascending=True).head()

city  elevation_ft
57    Calipatria        -180.0
463  Westmorland        -164.0
45       Brawley        -112.0
81     Coachella         -66.0
187     Imperial         -59.0

Let's move on to the cities with the highest and lowest population. We'll do this in the same way we analyzed the elevation of the cities, by sorting the data according to the population column. 

Here are the five cities with the highest population:

In [6]:
calidata[["city", "population_total"]].sort_values(by=["population_total"], ascending=False).head()

city  population_total
239    LosAngeles           3884307
367      SanDiego           1345895
375       SanJose           1000536
370  SanFrancisco            837442
150        Fresno            509039

Here are the five cities with the lowest population:

In [7]:
calidata[["city", "population_total"]].sort_values(by=["population_total"], ascending=True).head()

city  population_total
327      Pomona                 1
448      Vernon               112
7    AmadorCity               185
190    Industry               219
366    SandCity               334

But something doesn't seem quite right here. Is it really possible that the city of Pomona California has a population of 1? According to Google, the population of Pomona, CA is 151,348. That's a pretty big difference. Let's correct that value, and then reexamine the data. 

In [8]:
calidata.loc[(calidata.city== "Pomona"), "population_total"] = 151348

In [9]:
calidata[["city", "population_total"]].sort_values(by=["population_total"], ascending=True).head()

city  population_total
448      Vernon               112
7    AmadorCity               185
190    Industry               219
366    SandCity               334
435    Trinidad               367

How about Vernon California? According to Google, their population is 114, as of 2013. Since the data set we're working with does not provide a year, it's not unreasonable to presume that this data, though possibly slightly outdated, is correct. 

Finally, let's examine the cities with the greatest total area:

In [10]:
calidata[["city", "area_total_sq_mi"]].sort_values(by=["area_total_sq_mi"], ascending=False).head()

city  area_total_sq_mi
239      LosAngeles           503.000
367        SanDiego           372.400
370    SanFrancisco           231.890
55   CaliforniaCity           203.631
375         SanJose           179.970

A quick google search confirms that this is correct. 

# NYC Cause of Death Data Set

Let's move on to the Academic Journal data set. This data set contains information about the leading causes of death in New York City, by race and gender. 

In [11]:
deathdata = pd.read_csv("C:/Users/Jessica/Desktop/John's School/IS362/IS362_Project2/nyc_death_data.csv")
deathdata

Year                                      Leading Cause Sex  \
0     2014                        Diabetes Mellitus (E10-E14)   F   
1     2011          Cerebrovascular Disease (Stroke: I60-I69)   M   
2     2008              Malignant Neoplasms (Cancer: C00-C97)   M   
3     2010              Malignant Neoplasms (Cancer: C00-C97)   F   
4     2012          Cerebrovascular Disease (Stroke: I60-I69)   M   
5     2007  Mental and Behavioral Disorders due to Use of ...   M   
6     2011                                   All Other Causes   F   
7     2007       Chronic Lower Respiratory Diseases (J40-J47)   F   
8     2012  Essential Hypertension and Renal Diseases (I10...   F   
9     2009     Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
10    2011       Chronic Lower Respiratory Diseases (J40-J47)   M   
11    2008          Cerebrovascular Disease (Stroke: I60-I69)   F   
12    2013     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
13    2010          Cerebrovascular Disease (Stroke: I60-I69)   F   
14    2012     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
15    2012              Malignant Neoplasms (Cancer: C00-C97)   M   
16    2011          Cerebrovascular Disease (Stroke: I60-I69)   F   
17    2014  Congenital Malformations, Deformations, and Ch...   F   
18    2009                                   All Other Causes   F   
19    2014                        Diabetes Mellitus (E10-E14)   M   
20    2014  Accidents Except Drug Posioning (V01-X39, X43,...   M   
21    2012          Cerebrovascular Disease (Stroke: I60-I69)   F   
22    2009     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
23    2014  Essential Hypertension and Renal Diseases (I10...   F   
24    2008  Certain Conditions originating in the Perinata...   M   
25    2012  Mental and Behavioral Disorders due to Acciden...   M   
26    2014              Malignant Neoplasms (Cancer: C00-C97)   F   
27    2011  Mental and Behavioral Disorders due to Acciden...   F   
28    2014  Accidents Except Drug Posioning (V01-X39, X43,...   F   
29    2008          Cerebrovascular Disease (Stroke: I60-I69)   M   
...    ...                                                ...  ..   
1064  2011                          Parkinson's Disease (G20)   F   
1065  2012                          Alzheimer's Disease (G30)   F   
1066  2007              Malignant Neoplasms (Cancer: C00-C97)   M   
1067  2009                        Diabetes Mellitus (E10-E14)   M   
1068  2014                        Diabetes Mellitus (E10-E14)   M   
1069  2011              Malignant Neoplasms (Cancer: C00-C97)   M   
1070  2011     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
1071  2007       Chronic Lower Respiratory Diseases (J40-J47)   F   
1072  2014       Chronic Lower Respiratory Diseases (J40-J47)   M   
1073  2008     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
1074  2014       Chronic Lower Respiratory Diseases (J40-J47)   F   
1075  2010  Accidents Except Drug Posioning (V01-X39, X43,...   F   
1076  2008  Nephritis, Nephrotic Syndrome and Nephrisis (N...   F   
1077  2013              Malignant Neoplasms (Cancer: C00-C97)   M   
1078  2013  Certain Conditions originating in the Perinata...   F   
1079  2013  Mental and Behavioral Disorders due to Acciden...   M   
1080  2011                        Diabetes Mellitus (E10-E14)   F   
1081  2010                                   All Other Causes   M   
1082  2009                                   All Other Causes   F   
1083  2012  Accidents Except Drug Posioning (V01-X39, X43,...   F   
1084  2008                        Diabetes Mellitus (E10-E14)   M   
1085  2008     Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
1086  2007                                   All Other Causes   M   
1087  2013       Chronic Lower Respiratory Diseases (J40-J47)   M   
1088  2011              Malignant Neoplasms (Cancer: C00-C97)   M   
1089  2013  Essential Hypertension and Renal Diseases (I10...   F   
1090

As we can see, this data has a lot of null values. Unfortunately, instead of leaving the fields blank, there is a "." in those fields. First we will remove those characters so that the null values are reflected correctly. 

In [32]:
deathdata = deathdata.replace('.', np.nan)

In my initial analysis of the data, I noticed that the numerical values were sorting in a strange way, with 30 coming up as smaller than 4000. I realized this is because the numbers were not being read correctly. 

In [33]:
deathdata[['Deaths']] = deathdata[['Deaths']].apply(pd.to_numeric)
deathdata.dtypes

Year                         int64
Leading Cause               object
Sex                         object
Race Ethnicity              object
Deaths                     float64
Death Rate                  object
Age Adjusted Death Rate     object
dtype: object

Now let's ask some questions about this data:
• What was the leading cause of death for each demographic in 2007?
• What was the leading cause of death for each demographic in 2014?

Leading cause of death for Asian and Pacific Islanders in 2007:

In [34]:
deathdata[(deathdata['Year']==2007) & (deathdata['Race Ethnicity']=='Asian and Pacific Islander')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
781  2007           Malignant Neoplasms (Cancer: C00-C97)   M   
364  2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
404  2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
741  2007           Malignant Neoplasms (Cancer: C00-C97)   F   
483  2007                                All Other Causes   M   

                 Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
781  Asian and Pacific Islander   528.0      109.1                   145.9  
364  Asian and Pacific Islander   496.0      102.5                   157.8  
404  Asian and Pacific Islander   428.0       83.1                   115.2  
741  Asian and Pacific Islander   395.0       76.7                    88.4  
483  Asian and Pacific Islander   221.0       45.7                    61.2

Leading cause of death for Black Non-Hispanic in 2007:

In [35]:
deathdata[(deathdata['Year']==2007) & (deathdata['Race Ethnicity']=='Black Non-Hispanic')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
447  2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
65   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
461  2007           Malignant Neoplasms (Cancer: C00-C97)   F   
811  2007           Malignant Neoplasms (Cancer: C00-C97)   M   
715  2007                                All Other Causes   F   

         Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
447  Black Non-Hispanic  2722.0      258.6                   247.3  
65   Black Non-Hispanic  2121.0      248.9                   340.2  
461  Black Non-Hispanic  1800.0        171                   161.7  
811  Black Non-Hispanic  1523.0      178.7                   229.4  
715  Black Non-Hispanic  1230.0      116.8                   113.4

Leading cause of death for Hispanics in 2007:

In [36]:
deathdata[(deathdata['Year']==2007) & (deathdata['Race Ethnicity']=='Hispanic')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex Race Ethnicity  \
356   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F       Hispanic   
837   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M       Hispanic   
1066  2007           Malignant Neoplasms (Cancer: C00-C97)   M       Hispanic   
235   2007                                All Other Causes   M       Hispanic   
366   2007           Malignant Neoplasms (Cancer: C00-C97)   F       Hispanic   

      Deaths Death Rate Age Adjusted Death Rate  
356   1418.0      121.5                   164.3  
837   1327.0      121.7                   238.4  
1066  1013.0       92.9                   163.5  
235    988.0       90.6                   142.6  
366    969.0         83                   100.7

Leading cause of death for Asian and Unknown Races in 2007:

In [37]:
deathdata[(deathdata['Year']==2007) & (deathdata['Race Ethnicity']=='Not Stated/Unknown')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
618  2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
878  2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
769  2007           Malignant Neoplasms (Cancer: C00-C97)   F   
40   2007           Malignant Neoplasms (Cancer: C00-C97)   M   
716  2007                                All Other Causes   M   

         Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
618  Not Stated/Unknown    86.0        NaN                     NaN  
878  Not Stated/Unknown    82.0        NaN                     NaN  
769  Not Stated/Unknown    51.0        NaN                     NaN  
40   Not Stated/Unknown    45.0        NaN                     NaN  
716  Not Stated/Unknown    43.0        NaN                     NaN

Leading cause of death for Other in 2007:

In [38]:
deathdata[(deathdata['Year']==2007) & (deathdata['Race Ethnicity']=='Other Race/ Ethnicity')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
637   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
830   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
686   2007           Malignant Neoplasms (Cancer: C00-C97)   M   
1086  2007                                All Other Causes   M   
255   2007           Malignant Neoplasms (Cancer: C00-C97)   F   

             Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
637   Other Race/ Ethnicity    43.0        NaN                     NaN  
830   Other Race/ Ethnicity    36.0        NaN                     NaN  
686   Other Race/ Ethnicity    29.0        NaN                     NaN  
1086  Other Race/ Ethnicity    24.0        NaN                     NaN  
255   Other Race/ Ethnicity    22.0        NaN                     NaN

Leading cause of death for White Non-Hispanic in 2007:

In [39]:
deathdata[(deathdata['Year']==2007) & (deathdata['Race Ethnicity']=='White Non-Hispanic')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
833  2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
657  2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
476  2007           Malignant Neoplasms (Cancer: C00-C97)   F   
79   2007           Malignant Neoplasms (Cancer: C00-C97)   M   
272  2007                                All Other Causes   M   

         Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
833  White Non-Hispanic  7050.0      491.4                   250.7  
657  White Non-Hispanic  5632.0        421                   350.7  
476  White Non-Hispanic  3518.0      245.2                   167.4  
79   White Non-Hispanic  3356.0      250.9                   213.7  
272  White Non-Hispanic  1749.0      130.7                   115.9

Let's examine if and how these numbers changed by 2014:

Leading cause of death for Asian and Pacific Islanders in 2014:

In [40]:
deathdata[(deathdata['Year']==2014) & (deathdata['Race Ethnicity']=='Asian and Pacific Islander')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
210  2014           Malignant Neoplasms (Cancer: C00-C97)   M   
988  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
344  2014           Malignant Neoplasms (Cancer: C00-C97)   F   
567  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
54   2014                                All Other Causes   M   

                 Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
210  Asian and Pacific Islander   657.0      114.5                   129.5  
988  Asian and Pacific Islander   554.0       96.5                   118.5  
344  Asian and Pacific Islander   502.0       80.2                    80.6  
567  Asian and Pacific Islander   462.0       73.8                    81.1  
54   Asian and Pacific Islander   424.0       73.9                    90.4

Leading cause of death for Black Non-Hispanic in 2014:

In [41]:
deathdata[(deathdata['Year']==2014) & (deathdata['Race Ethnicity']=='Black Non-Hispanic')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
819  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
303  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
231  2014           Malignant Neoplasms (Cancer: C00-C97)   F   
810  2014                                All Other Causes   F   
278  2014           Malignant Neoplasms (Cancer: C00-C97)   M   

         Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
819  Black Non-Hispanic  2194.0      209.1                   169.1  
303  Black Non-Hispanic  1958.0      226.8                   264.7  
231  Black Non-Hispanic  1852.0      176.5                   148.4  
810  Black Non-Hispanic  1536.0      146.4                   126.4  
278  Black Non-Hispanic  1532.0      177.5                   199.6

Leading cause of death for Hispanics in 2014:

In [42]:
deathdata[(deathdata['Year']==2014) & (deathdata['Race Ethnicity']=='Hispanic')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex Race Ethnicity  \
517  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M       Hispanic   
908  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F       Hispanic   
304  2014                                All Other Causes   M       Hispanic   
526  2014           Malignant Neoplasms (Cancer: C00-C97)   F       Hispanic   
271  2014           Malignant Neoplasms (Cancer: C00-C97)   M       Hispanic   

     Deaths Death Rate Age Adjusted Death Rate  
517  1281.0      107.3                   170.5  
908  1230.0       97.1                   106.7  
304  1195.0      100.1                   143.3  
526  1154.0       91.1                    97.4  
271  1146.0         96                   143.5

Leading cause of death for Asian and Unknown Races in 2014:

In [43]:
deathdata[(deathdata['Year']==2014) & (deathdata['Race Ethnicity']=='Not Stated/Unknown')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
277  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
127  2014                                All Other Causes   M   
603  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
355  2014           Malignant Neoplasms (Cancer: C00-C97)   M   
314  2014           Malignant Neoplasms (Cancer: C00-C97)   F   

         Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
277  Not Stated/Unknown   115.0        NaN                     NaN  
127  Not Stated/Unknown    96.0        NaN                     NaN  
603  Not Stated/Unknown    95.0        NaN                     NaN  
355  Not Stated/Unknown    73.0        NaN                     NaN  
314  Not Stated/Unknown    50.0        NaN                     NaN

Leading cause of death for Other in 2014:

In [44]:
deathdata[(deathdata['Year']==2014) & (deathdata['Race Ethnicity']=='Other Race/ Ethnicity')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
264  2014                                All Other Causes   M   
612  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
26   2014           Malignant Neoplasms (Cancer: C00-C97)   F   
898  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
251  2014                                All Other Causes   F   

            Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
264  Other Race/ Ethnicity    74.0        NaN                     NaN  
612  Other Race/ Ethnicity    68.0        NaN                     NaN  
26   Other Race/ Ethnicity    66.0        NaN                     NaN  
898  Other Race/ Ethnicity    63.0        NaN                     NaN  
251  Other Race/ Ethnicity    59.0        NaN                     NaN

Leading cause of death for White Non-Hispanic in 2014:

In [45]:
deathdata[(deathdata['Year']==2014) & (deathdata['Race Ethnicity']=='White Non-Hispanic')].sort_values('Deaths', ascending=False).head()

Year                                   Leading Cause Sex  \
620  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
633  2014  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
953  2014           Malignant Neoplasms (Cancer: C00-C97)   F   
607  2014           Malignant Neoplasms (Cancer: C00-C97)   M   
726  2014                                All Other Causes   F   

         Race Ethnicity  Deaths Death Rate Age Adjusted Death Rate  
620  White Non-Hispanic  4507.0        318                     161  
633  White Non-Hispanic  3990.0      297.1                   238.4  
953  White Non-Hispanic  3153.0      222.5                   150.2  
607  White Non-Hispanic  3142.0        234                   195.1  
726  White Non-Hispanic  2578.0      181.9                   110.7

# International Student Data

This data set has a lot of information about international students, including their country of origin, their gender, their age, their dominant hand, their height, shoe size, and armspan, how many languages they speak, how they get to school, and how long it takes them to do so, their reaction time, and their favorite sport. 

In [46]:
studentdata = pd.read_csv("C:/Users/Jessica/Desktop/John's School/IS362/IS362_Project2/student_data.csv", encoding='latin-1')
studentdata

Country                        Region  Gender  Age-years  \
0      Australia               New South Wales  Female         10   
1      Australia                      Victoria  Female         14   
2    New Zealand                      Auckland  Female         14   
3             UK                    South West  Female         15   
4      Australia               New South Wales  Female         16   
5      Australia             Western Australia    Male         11   
6         Canada                       Alberta  Female         12   
7      Australia                    Queensland    Male         15   
8         Canada                       Ontario  Female         14   
9             UK                 West Midlands    Male         13   
10     Australia                      Tasmania  Female         11   
11   New Zealand                      Auckland    Male         10   
12   New Zealand                      Auckland  Female         10   
13        Canada                       Ontario    Male         12   
14        Canada                       Ontario    Male         17   
15     Australia                      Victoria  Female          7   
16            UK                 East Midlands  Female         11   
17        Canada                       Ontario  Female         13   
18        Canada                       Ontario  Female         12   
19            UK                    North West  Female         11   
20        Canada                       Alberta  Female         15   
21        Canada                       Ontario    Male         12   
22        Canada                       Ontario    Male         14   
23        Canada                       Ontario    Male         14   
24     Australia                      Tasmania    Male         16   
25            UK                 Home Counties    Male         14   
26        Canada                       Ontario  Female         16   
27     Australia  Australian Capitol Territory  Female          8   
28            UK                    North West  Female         15   
29   New Zealand                      Auckland  Female         13   
..           ...                           ...     ...        ...   
470          USA                            GA    Male         11   
471          USA                            MI  Female         17   
472          USA                            SC    Male         17   
473          USA                            DC    Male         16   
474          USA                            MO  Female         17   
475          USA                            PA  Female         17   
476          USA                            SC  Female         16   
477          USA                            SC    Male         17   
478          USA                            NV    Male         13   
479          USA                            NM  Female         17   
480          USA                            OK  Female         15   
481          USA                            CA    Male         17   
482          USA                            CA  Female         16   
483          USA                            SC    Male         18   
484          USA                            KY    Male         17   
485          USA                            CA  Female         11   
486          USA                            GA    Male         16   
487          USA                            AZ  Female         16   
488          USA                            SC    Male         15   
489          USA                            NH    Male         17   
490          USA                            CA  Female         17   
491          USA                            CO  Female         15   
492          USA                            CA  Female         17   
493          USA                            GA    Male         17   
494          USA                            PA    Male         16   
495          USA                            IN    Male         18   
496    

There are two problems with this data: one is that some of the values appear to be missing, or incorrect. Another is that there is so much data contained in this data set that it becomes difficult to focus in on the specifics. In order to focus in on certain areas of the data, let's ask targeted questions about this data that focuses on more narrow criteria:
• Are more students right handed or left handed?
• What is the most popular sport among students of each country?
• What is the average time it takes a student to get to school in each country?
• In which country are students most concerned about reducing pollution?

Are more students right handed or left handed?

In [47]:
studentdata.groupby('Country')['Handed'].value_counts()

Country      Handed      
Australia    Right-handed    141
             Left-handed      16
             Ambidextrous     13
Canada       Right-handed     88
             Ambidextrous      8
             Left-handed       4
New Zealand  Right-handed     61
             Left-handed       9
             Ambidextrous      4
UK           Right-handed     46
             Left-handed       9
             Ambidextrous      1
USA          Right-handed     89
             Left-handed       7
             Ambidextrous      3
Name: Handed, dtype: int64

In all five of the countries surveyed, there are more right handed students than left handed students. 

What is the most popular sport among students of each country?

In [48]:
studentdata.groupby('Country')['Favourite_physical_activity'].value_counts()

Country    Favourite_physical_activity
Australia  Football/Soccer                33
           Netball                        22
           Other activities/sports        22
           Dancing                        14
           Basketball                     10
           Hockey (Field)                 10
           Tennis                          9
           Rugby Union                     6
           Cricket                         5
           Swimming                        5
           Gymnastics                      4
           Martial arts                    4
           Rugby League                    4
           Athletics                       3
           Baseball/Softball               3
           Cycling                         3
           Table Tennis                    3
           Walking/Hiking                  3
           Skateboarding/Rollerblading     2
           Bowling                         1
           Golf                            1
           Hocke

What is the average time it takes a student to get to school in each country?

In [49]:
studentdata.groupby('Country')['Travel_time_to_School'].mean().sort_values(ascending=False)

Country
UK             19.964286
New Zealand    17.297297
Canada         16.950000
USA            16.885417
Australia      15.923529
Name: Travel_time_to_School, dtype: float64

Students in the UK have the longest commute to school on average, at nearly 20 minutes. Students in Australia have the shortest commute on average, with nearly 16 minutes. 

In which country are students most concerned about reducing pollution?

In [50]:
studentdata.groupby('Country')['Importance_reducing_pollution'].mean().sort_values(ascending=False)

Country
USA            886.375000
Australia      712.538922
New Zealand    702.702703
Canada         696.480000
UK             578.196429
Name: Importance_reducing_pollution, dtype: float64

According to the importance placed on reducing pollution, on average, students in the US are the most concerned about reducing pollution. Students in the UK are the least concerned about this issue. 